src/pkg.js

```diff
@@ -57,20 +55,23 @@ const updateVersion = ({ path, packageName, logger }) => (
             peerDependencies = {}
         }, version) => {
             if (dependencies.hasOwnProperty(packageName)) {
+                assert(!satisfies(version, dependencies[packageName]), `${packageName} latest version ${version} matches existing dependency ${dependencies[packageName]}`);
```

Comment:
This code base uses exceptions as control flow a lot (I know I know)...so an assert just blows up a single instance of an upgrade. So any time we want to stop the upgrade, we just throw.